modules/publication/ajax/FileUpload.php

```diff
@@ -100,6 +100,15 @@ function uploadPublication() : void
     $db->insert('publication', $fields);
     $pubID = $db->getLastInsertId();
 
+    // give creator permission to edit
+    $db->insert(
+        'publication_users_edit_perm_rel',
+        [
+            'PublicationID' => $pubID,
+            'UserID'        => $uid,
+        ]
+    );
+
```

Comment:
I think admin should always be added to users with edit permissions

```suggestion
    $db->insert(
        'publication_users_edit_perm_rel',
        [
            'PublicationID' => $pubID,
            'UserID'        => $uid,
        ]
    );


    $db->insert(
        'publication_users_edit_perm_rel',
        [
            'PublicationID' => $pubID,
            'UserID'        => 'admin',
        ]
    );
```

Reply:
`admin` is only a CCNA username. Other projects may have different names. For example I believe raisinbread uses `lorisadmin`. Also `admin` is not the `userID`.

Should I just add `userID => 1`? Is this good practice ? @ridz1208

Reply:
correct, but also, we don't really want to automatically assign admin permissions in that way. If the goal is to give admin access to everything then this should be handled where you check for the permissions not where you assign the permissions (and even at that I don't think admin should be able to have access and edit everything).